multicluster/cmd/link.go

```diff
@@ -200,75 +201,92 @@ A full list of configurable values can be found at https://github.com/linkerd/li
 				return err
 			}
 
-			gateway, err := k.CoreV1().Services(opts.gatewayNamespace).Get(cmd.Context(), opts.gatewayName, metav1.GetOptions{})
+			remoteDiscoverySelector, err := metav1.ParseToLabelSelector(opts.remoteDiscoverySelector)
 			if err != nil {
 				return err
 			}
 
-			gatewayAddresses := ""
-			gwAddresses := []string{}
-			for _, ingress := range gateway.Status.LoadBalancer.Ingress {
-				addr := ingress.IP
-				if addr == "" {
-					addr = ingress.Hostname
-				}
-				if addr == "" {
-					continue
+			var link mc.Link
+			if opts.disableGateway {
+				// When multicluster has been installed without a gateway, and link
+				// has been explicitly invoked with `--no-gateway`, then build a
+				// link with no gateway information, and a "nothing" selector.
+				link = mc.Link{
```

Comment:
is it possible to build this common part of the Link struct up above this if branch and then populate the gateway parts of it only when appropriate? to avoid duplication.